src/query/stream/streams.cpp

```diff
@@ -639,6 +639,18 @@ void Streams::Start(const std::string &stream_name) {
       it->second);
 }
 
+void Streams::StartWithLimit(const std::string &stream_name, int64_t batch_limit) const {
+  auto locked_streams = streams_.ReadLock();
+  auto it = GetStream(*locked_streams, stream_name);
+
+  std::visit(
+      [&](auto &&stream_data) {
+        auto stream_source_ptr = stream_data.stream_source->ReadLock();
+        stream_source_ptr->StartWithLimit(batch_limit);
```

Comment:
```suggestion
        locked_streams.reset();
        stream_source_ptr->StartWithLimit(batch_limit);
```

You have to release the lock on `streams_`, otherwise it might block deleting/adding other streams. See how `Check` is implemented.